src/main/java/com/google/api/codegen/transformer/csharp/CSharpSampleImportTransformer.java

```diff
@@ -0,0 +1,81 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.csharp;
+
+import com.google.api.codegen.config.MethodContext;
+import com.google.api.codegen.config.MethodModel;
+import com.google.api.codegen.metacode.InitCodeNode;
+import com.google.api.codegen.transformer.ImportTypeTable;
+import com.google.api.codegen.transformer.StandardImportSectionTransformer;
+import com.google.api.codegen.transformer.StandardSampleImportTransformer;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.codegen.viewmodel.CallingForm;
+
+public class CSharpSampleImportTransformer extends StandardSampleImportTransformer {
+
+  public CSharpSampleImportTransformer() {
+    super(new StandardImportSectionTransformer());
+  }
+
+  @Override
+  protected void addSampleBodyImports(MethodContext context, CallingForm form) {
+    ImportTypeTable typeTable = context.getTypeTable();
+    MethodModel method = context.getMethodModel();
+    SurfaceNamer namer = context.getNamer();
+    if (namer.usesAsyncAwaitPattern(form)) {
+      typeTable.saveNicknameFor("System.Threading.Tasks.Task");
+    }
+    switch (form) {
+      case RequestPaged:
+      case FlattenedPaged:
+      case RequestPagedAll:
+      case FlattenedPagedAll:
+        typeTable.saveNicknameFor("Google.Api.Gax.PagedEnumerable");
+        break;
+      case RequestAsyncPaged:
+      case FlattenedAsyncPaged:
+      case RequestAsyncPagedAll:
+      case FlattenedAsyncPagedAll:
+        typeTable.saveNicknameFor("Google.Api.Gax.PagedAsyncEnumerable");
+        break;
+      case RequestPagedPageSize:
+      case FlattenedPagedPageSize:
+        typeTable.saveNicknameFor("Google.Api.Gax.Page");
+        break;
+    }
+  }
+
+  @Override
+  protected void addInitCodeImports(
+      MethodContext context, ImportTypeTable initCodeTypeTable, Iterable<InitCodeNode> nodes) {
+    super.addInitCodeImports(context, initCodeTypeTable, nodes);
+    ImportTypeTable typeTable = context.getTypeTable();
+    for (InitCodeNode node : nodes) {
+      typeTable.getAndSaveNicknameFor(node.getType());
+      switch (node.getLineType()) {
+        case ReadFileInitLine:
+          typeTable.saveNicknameFor("System.IO.File");
+          break;
+        case SimpleInitLine:
+        case StructureInitLine:
+        case ListInitLine:
+        case MapInitLine:
+          break; // fall through
```

Comment:
`break; // fall through` is confusing

You probably want each of the lines above to have the comment:
```
case SimpleInitLine:  // fall-through
...
caseMapInitLine:
  break;
```

Reply:
Actually, found a better way using `Stream`